android/anc/src/main/java/org/smartregister/fhircore/anc/ui/anccare/details/AncDetailsViewModel.kt

```diff
@@ -46,15 +50,74 @@ class AncDetailsViewModel(
     return patientDemographics
   }
 
-  fun fetchCarePlan(cJson: String?): LiveData<List<CarePlanItem>> {
+  fun fetchCarePlan(): LiveData<List<CarePlanItem>> {
     val patientCarePlan = MutableLiveData<List<CarePlanItem>>()
     viewModelScope.launch(dispatcher.io()) {
-      val listCarePlan = ancPatientRepository.fetchCarePlan(patientId = patientId, cJson)
-      patientCarePlan.postValue(listCarePlan)
+      val listCarePlan = ancPatientRepository.searchCarePlan(id = patientId)
+      val listCarePlanItem = ancPatientRepository.fetchCarePlanItem(listCarePlan, patientId)
+      patientCarePlan.postValue(listCarePlanItem)
     }
     return patientCarePlan
   }
 
+  fun fetchObservation(): LiveData<AncOverviewItem> {
+    val patientAncOverviewItem = MutableLiveData<AncOverviewItem>()
+    val ancOverviewItem = AncOverviewItem()
+    viewModelScope.launch(dispatcher.io()) {
+      val listObservation = ancPatientRepository.fetchObservations(patientId = patientId)
+      if (listObservation.isNotEmpty()) {
+        for (i in listObservation.indices) {
+          if (listObservation[i].code != null)
+            if (listObservation[i].code.coding != null)
+              if (listObservation[i].code.coding.isNotEmpty())
```

Comment:
I think these can be in a single conjunction b/c it'll execute in order and short circuit if any fail